packages/browser/src/node/pool.ts

```diff
@@ -37,6 +37,23 @@ export function createBrowserPool(ctx: Vitest): ProcessPool {
       )
     }
 
+    async function setBreakpoint(contextId: string, file: string) {
+      if (!project.config.inspector.waitForDebugger) {
+        return
+      }
+
+      if (!provider.getCDPSession) {
+        throw new Error('Unable to set breakpoint, CDP not supported')
+      }
+
+      const session = await provider.getCDPSession(contextId)
+      await session.send('Debugger.enable', {})
+      await session.send('Debugger.setBreakpointByUrl', {
+        lineNumber: 0, // TODO: Need to make sure test files have source map location on first line
+        urlRegex: escapePathToRegexp(file),
+      })
```

Comment:
This one doesn't yet work. The demo video on description is recorded by using `lineNumber: 3` here.

Some solutions I've already tried without much luck but might we worth to retry with more time:

- Try forcing source map location in `packages/browser/src/node/plugin.ts
` like we do on Node side:

https://github.com/vitest-dev/vitest/blob/372a2c26a8e5ac2cfe6db961ed40529f09464e82/packages/vite-node/src/source-map.ts#L64-L68

- Or try using `Debugger.getPossibleBreakpoints` and `Debugger.setBreakpoint`. At first try this didn't work, as `Debugger.getPossibleBreakpoints` returned positions of **transpiled module** - we want to set breakpoints for the remapped file instead.

Reply:
Using `urlRegex` instead of `url` as the files have query paramters like `?browser` etc.

Reply:
It was timing issue! Breakpoint has to be set before page is navigated to.